File: bundle_code.py
import os

def bundle_code():
    output_file = 'codebase_context.txt'
    directories_to_scan = ['lib']
    files_to_include = ['pubspec.yaml', 'README.md']
    
    with open(output_file, 'w', encoding='utf-8') as outfile:
        # Write individual files first
        for file_name in files_to_include:
            if os.path.exists(file_name):
                outfile.write(f"--- File: {file_name} ---\n")
                with open(file_name, 'r', encoding='utf-8') as f:
                    outfile.write(f.read())
                outfile.write("\n\n")
        
        # Walk through directories and write .dart files
        for directory in directories_to_scan:
            for root, _, files in os.walk(directory):
                for file in files:
                    if file.endswith('.dart'):
                        file_path = os.path.join(root, file)
                        outfile.write(f"--- File: {file_path} ---\n")
                        try:
                            with open(file_path, 'r', encoding='utf-8') as f:
                                outfile.write(f.read())
                        except Exception as e:
                            outfile.write(f"Error reading file: {e}\n")
                        outfile.write("\n\n")

if __name__ == '__main__':
    bundle_code()
    print("Codebase successfully bundled into codebase_context.txt!")

File: bundle_code.dart
import 'dart:io';

void main() async {
  final outputFile = File('codebase_context.txt');
  final sink = outputFile.openWrite();

  final filesToInclude = ['pubspec.yaml', 'README.md'];
  final directoriesToScan = ['lib'];

  for (final fileName in filesToInclude) {
    final file = File(fileName);
    if (await file.exists()) {
      sink.writeln('--- File: $fileName ---');
      final content = await file.readAsString();
      sink.writeln(content);
      sink.writeln('\n');
    }
  }

  for (final dirName in directoriesToScan) {
    final dir = Directory(dirName);
    if (await dir.exists()) {
      await for (final entity in dir.list(recursive: true)) {
        if (entity is File && entity.path.endsWith('.dart')) {
          sink.writeln('--- File: ${entity.path} ---');
          try {
            final content = await entity.readAsString();
            sink.writeln(content);
          } catch (e) {
            sink.writeln('Error reading file: $e');
          }
          sink.writeln('\n');
        }
      }
    }
  }

  await sink.close();
  print('Codebase successfully bundled into codebase_context.txt!');
}
